### PR TITLE
Stop List View #29

### PR DESCRIPTION
**Pull request by raphaelmue**
See #29

### CODE REVIEW

File: android/app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -25,4 +25,17 @@
     <string name="error_no_city_found">Keine Stadt mit dem Namen \"%1$s\" gefunden.</string>
     <string name="user">Benutzer</string>
     <string name="search">Suche</string>
+    <string name="add">Hinzufügen</string>
+    <string name="hours_unit">%1$s std</string>
+    <string name="rate">Bewerten</string>
+    <string name="number_of_stop_found">%d Sehenswürdigkeiten gefunden!</string>
+    <string name="category_place_of_worship">Gotteshaus</string>
```
Comment (joshuaschu) [inline android/app/src/main/res/values-de/strings.xml]: Maybe something like "Religiöse Versammlungsstätte"... I mean there are religions that are praying to the pasta monster....

File: java/org.travlyn.server/src/main/java/org/travlyn/server/externalapi/access/DBPediaCityRequest.java [hunk 2]
```diff
@@ -4,33 +4,30 @@
...
 import java.util.Set;
 
 /**
  * Utility class for querying the DBmedia API for a specified city.
  *
  * @author Joshua Schulz
  */
-public class DBpediaCityRequest implements DBpediaRequest<City> {
-    private static final String BASE_API = "http://vmdbpedia.informatik.uni-leipzig.de:8080/api/1.0.0/values";
+public class DBPediaCityRequest extends DBPediaRequest<City> {
```
Comment (joshuaschu) [inline java/org.travlyn.server/src/main/java/org/travlyn/server/externalapi/access/DBPediaCityRequest.java]: I guess "DBpedia" is the official naming... i guess we should stay with that

File: java/org.travlyn.server/src/main/java/org/travlyn/server/service/TravlynService.java [hunk 3]
```diff
@@ -80,15 +80,20 @@ public City getCityWithInformation(String city) {
                     .getSingleResult();
 
             //return cached city
+            //City city1 = entity.toDataTransferObject();
+            //city1.toEntity();
             return entity.toDataTransferObject();
         } catch (NoResultException noResult) {
             // city is not cached --> get from api
-            DBpediaCityRequest request = new DBpediaCityRequest(city);
+            DBPediaCityRequest request = new DBPediaCityRequest(city);
```
Comment (joshuaschu) [inline java/org.travlyn.server/src/main/java/org/travlyn/server/service/TravlynService.java]: see above..

File: java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/DBPediaAccessTest.java [hunk 4]
```diff
@@ -6,10 +6,10 @@
 import org.travlyn.shared.model.api.City;
 
 @Tag("unit")
-public class DBpediaAccessTest {
+public class DBPediaAccessTest {
```
Comment (joshuaschu) [inline java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/DBPediaAccessTest.java]: see above

File: java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/OpenRouteStopAccessTest.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.Test;
+import org.junit.jupiter.api.Tag;
+import org.travlyn.shared.model.api.Stop;
+
+@Tag("unit")
+public class OpenRouteStopAccessTest {
+
+    @Test
+    public void testStopCall(){
+        //OpenrouteRequest request = new OpenrouteRequest();
```
Comment (joshuaschu) [inline java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/OpenRouteStopAccessTest.java]: Souldn't we include this test? Who is responsible for fixing... you or me?

File: java/org.travlyn.shared/src/main/java/org/travlyn/shared/model/db/StopEntity.java [hunk 6]
```diff
@@ -33,11 +40,15 @@
...
+    @ManyToOne
+    @JoinColumn(name="city_id", referencedColumnName = "id")
+    private CityEntity city;
+
     @OneToMany
     @JoinColumn(name = "ratable")
     private Set<StopRatingEntity> ratings;
 
-    @OneToOne(targetEntity = CategoryEntity.class)
+    @OneToOne(targetEntity = CategoryEntity.class, cascade = {CascadeType.ALL}, orphanRemoval = true)
```
Comment (joshuaschu) [inline java/org.travlyn.shared/src/main/java/org/travlyn/shared/model/db/StopEntity.java]: Should we really remove the category if a stop is deleted?

File: android/app/src/main/java/org/travlyn/ui/home/HomeFragment.kt [hunk 7]
```diff
@@ -218,6 +193,36 @@ class HomeFragment : Fragment() {
         }
     }
 
+    private fun addCityToSuggestions(city: City) {
```
[the remark this answers is not in the document]
Comment (raphaelmue) [reply android/app/src/main/java/org/travlyn/ui/home/HomeFragment.kt]: See #42

File: java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/OpenRouteStopAccessTest.java (hunk 5 above)
Comment (raphaelmue) [reply java/org.travlyn.server/src/test/java/org/travlyn/server/externalapi/access/OpenRouteStopAccessTest.java]: Both tests are fixed and improved

**Pull request merged**